src/store/State.ts

```diff
@@ -35,6 +35,7 @@ export interface State {
   hub?: Hub;
   // Top-level state items, keep these to a minimum
   loading: boolean;
```

Comment:
note that we also have a loading state item. so any "loading" component you wanna use can also use that